src/cuda/my_kernel.cu

```diff
@@ -581,6 +581,12 @@ extern "C" float cuda_block_wrapper(uint dim, uint* grid, uint* block,
         double *final_gather_data,
         char validate)
 {
+
+    if (pat_len > 1024) {
```

Comment:
Would it be better to compare the pattern length to the local work size, instead of 1024? 1024 is the maximum for local work size on most NVidia platforms, but if the user were to set it to 256 and try to run a pattern length of 1024, this wouldn't catch that case.